### PR TITLE
Update for issue 77.

### PR DESCRIPTION
**Pull request by ghost**
Fix as described in issue 77.

Prevents crash in LIBZMQ encoder.cpp, by using a unique message for every send and receive.

### CODE REVIEW

File: src/clrzmq/Socket.cs [hunk 1]
```diff
@@ -473,26 +468,45 @@ private static string GetTransportName(Transport tr)
         /// Forward all message parts directly to destination. No marshalling performed.
         /// </summary>
         /// <param name="destination">Destination Socket</param>
-        public void Forward(Socket destination) {
-            if (destination == null) {
+        public void Forward(Socket destination)
+        {
+            IntPtr _msg;
+            _msg = Marshal.AllocHGlobal(ZMQ_MSG_T_SIZE);
```
Comment (jgoz) [inline src/clrzmq/Socket.cs:474]: Allocating here will result in a memory leak if `destination` turns out to be null.

Comment (ghost) [inline src/clrzmq/Socket.cs:474]: Ahh so then I should move this below the destination check. Good catch.

Comment (ghost) [inline src/clrzmq/Socket.cs:474]: Do I need to make a new pull for the change or is there a way for me to update this commit?

Comment (ghost) [inline src/clrzmq/Socket.cs:474]: I've checked and I've also found similar issues with the send and receive functions. Those should all be fixed. just let me know how you want me to proceed and I'll get you those changes.

**Comment by jgoz** [comment]
Hi Aaron, this pull request is much easier to follow. Thanks for that.

The idea here is correct, but there are a few memory cleanup issues that could be solved by using a `DisposableIntPtr` (from Util.cs) for `msg`. I will modify your changeset to use that instead. I'll also commit a similar fix to the master branch as the issue almost certainly exists there too.

File: src/clrzmq/Socket.cs (hunk 1 above)
Comment (jgoz) [inline src/clrzmq/Socket.cs:474]: Please see my comment at the pull request level. You can add commits to this branch and they will appear in the pull request.

Comment (ghost) [inline src/clrzmq/Socket.cs:474]: Ahh, I see. Thank you very much.

**Comment by ghost** [comment]
"Hi Aaron, this pull request is much easier to follow. Thanks for that.

The idea here is correct, but there are a few memory cleanup issues that could be solved by using a DisposableIntPtr (from Util.cs) for msg. I will modify your changeset to use that instead. I'll also commit a similar fix to the master branch as the issue almost certainly exists there too."

Excellent glad to hear that. If I understand you correctly, your going to modify the code and I don't need to make any additional changes? Including what is in the diff discussion?

**Comment by jgoz** [comment]
Yes, I'll add my own changes, using yours as a starting point.

**Comment by ghost** [comment]
Fantastic. Thanks for all your work! :)

**Pull request merged**